native/src/main/java/io/ballerina/lib/avro/Avro.java

```diff
@@ -0,0 +1,112 @@
+/*
+ * Copyright (c) 2024, WSO2 LLC. (http://www.wso2.com)
+ *
+ * WSO2 LLC. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package io.ballerina.lib.avro;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.dataformat.avro.AvroMapper;
+import com.fasterxml.jackson.dataformat.avro.AvroSchema;
+import io.ballerina.runtime.api.creators.ValueCreator;
+import io.ballerina.runtime.api.utils.JsonUtils;
+import io.ballerina.runtime.api.utils.StringUtils;
+import io.ballerina.runtime.api.utils.ValueUtils;
+import io.ballerina.runtime.api.values.BArray;
+import io.ballerina.runtime.api.values.BHandle;
+import io.ballerina.runtime.api.values.BObject;
+import io.ballerina.runtime.api.values.BString;
+import io.ballerina.runtime.api.values.BTypedesc;
+import org.apache.avro.Schema;
+
+import java.io.IOException;
+import java.util.Objects;
+
+import static io.ballerina.lib.avro.ModuleUtils.AVRO_SCHEMA;
+
+public class Avro {
+
+    private Schema nativeSchema;
+
+    public Avro(BString schema) {
+        Schema.Parser parser = new Schema.Parser();
+        this.nativeSchema = parser.parse(schema.getValue());
+    }
+
+    protected Schema getSchema() {
+        return nativeSchema;
+    }
+
+    protected void setSchema(Schema nativeSchema) {
+        this.nativeSchema = nativeSchema;
+    }
+
+    public static Object toAvro(BObject avroObject, Object data) {
+        BHandle handle = (BHandle) avroObject.get(StringUtils.fromString(AVRO_SCHEMA));
+        Avro avro = (Avro) handle.getValue();
+        Schema schema = avro.getSchema();
+        ObjectMapper objectMapper = new ObjectMapper();
+        Object jsonObject;
+        try {
+            jsonObject = generateJsonObject(data, schema, objectMapper);
```

Comment:
I think we can do this instead. 
```suggestion
        try {
            Object jsonObject = generateJsonObject(data, schema, objectMapper);
```